src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_LENGTH = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[START_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        if (!enoughSpace()) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (!correctIndexForAdd(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Cant add element at such index");
+        }
+        if (!enoughSpace()) {
+            grow();
+        }
+        if (index != size) {
+            for (int i = size - 1; i >= index; i--) {
+                array[i + 1] = array[i];
+            }
+        }
```

Comment:
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html): Use [System.arraycopy()](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#arraycopy-java.lang.Object-int-java.lang.Object-int-int-) to move your array elements.